packages/core/content-releases/server/src/controllers/release.ts

```diff
@@ -29,50 +28,112 @@ const releaseController = {
 
     const releaseService = getService('release', { strapi });
 
-    // Handle requests for releases filtered by content type entry
-    const isFindManyForContentTypeEntry = Boolean(ctx.query?.contentTypeUid && ctx.query?.entryId);
-    if (isFindManyForContentTypeEntry) {
-      const query: GetContentTypeEntryReleases.Request['query'] =
-        await permissionsManager.sanitizeQuery(ctx.query);
+    const query = await permissionsManager.sanitizeQuery(ctx.query);
+    const { contentTypeUid, documentId, hasEntryAttached, locale } = query;
+    const entryId = await getEntryId({ contentTypeUid, documentId, locale }, { strapi });
 
-      const contentTypeUid = query.contentTypeUid;
-      const entryId = query.entryId;
-      // Parse the string value or fallback to a default
-      const hasEntryAttached: GetContentTypeEntryReleases.Request['query']['hasEntryAttached'] =
-        typeof query.hasEntryAttached === 'string' ? JSON.parse(query.hasEntryAttached) : false;
-
-      const data = hasEntryAttached
-        ? await releaseService.findManyWithContentTypeEntryAttached(contentTypeUid, entryId)
-        : await releaseService.findManyWithoutContentTypeEntryAttached(contentTypeUid, entryId);
-
-      ctx.body = { data };
-    } else {
-      const query: GetReleases.Request['query'] = await permissionsManager.sanitizeQuery(ctx.query);
-      const { results, pagination } = await releaseService.findPage(query);
+    if (!entryId) {
+      throw new errors.NotFoundError('Entry not found');
+    }
 
-      const data = results.map((release: ReleaseWithPopulatedActions) => {
-        const { actions, ...releaseData } = release;
+    const isEntryAttachedBoolean =
+      typeof hasEntryAttached === 'string' ? JSON.parse(hasEntryAttached) : false;
 
-        return {
-          ...releaseData,
+    if (isEntryAttachedBoolean) {
+      const releases = await releaseService.findMany({
+        where: {
           actions: {
-            meta: {
-              count: actions.count,
+            target_type: contentTypeUid,
+            target_id: entryId,
```

Comment:
That's a good point and probably is going to be a problem 🤔

I'm using the entryId because I think polymorphic relations with documentId are not implemented yet, right? So I don't see a solution 🤔

Reply:
we might need to find alternatives, relations will allways be stored using entry ids, it's just that we expose docId relations to make things consistent on the api.

I guess, at the moment, discarding a draft would result in removing that one from the release 🤔

Reply:
Yes, we are using a db lifecycle to listen to that "event" and remove deleted entries from releases

Reply:
well , I guess discarding the documetn would just unlist it from the release then.
It might be a good discussion with Yannis to check if this is an issue and if we want to work on it on the future , wdyt?